### ISSUE TITLE
Configure download folder (external_sd)

### ISSUE THREAD

**Issue by mokasin** [owncloud/android#10]
At the moment with version 1.3.12 on Android 2.3.6 the ownCloud  client seems to sync with a hardcoded folder at /sdcard/owncloud.

On my phone /sdcard points to the internal memory of the phone, which is quiet small.

Please add an option to set the syncing target directory in the client.

**Comment by Kernald** [comment]
Is there any behaviour decided on what to do with existing files? Is there any path stored in the database?

**Comment by davivel** [comment]
@Kernald, paths are stored in database for every file with local copy, and probably should remain that way, since we want to change the current behaviour in uploads and stop duplicating files - what means keeping a full path at least for the uploaded files.

This issue is considered in our roadmap, but without no estimated time to implement it. In few words: we need to change the implementation of ```FileStorageUtils.getSavePath()``` so that the current base ```sdCard.getAbsolutePath() + "/" + MainApp.getDataFolder() + "/"``` is changed with a value previously saved in the app settings. Its default value should match ```sdCard.getAbsolutePath() + "/" + MainApp.getDataFolder() + "/"```. 

Then we would need adding an entry in the Settings activity to allow the users change the value saved in settings. Changing the value would require ** moving ** the current folder to the new location, what may result in errors to condider, of course.

**Comment by Kernald** [comment]
Okay, thanks. Moving files involve updating the database entries, hence my question.

**Comment by tomachinz** [comment]
I would like to second this - its a VITAL feature that I am not installing Bittorrent Sync in order to get arond: BTSync is able to sync both /mnt/sdcard (internal) and also /mnt/extsdcard (external) even if both are encrypted.
Very frustrating to have to go back to BTSync when I've nearly switched everything else over to ownCloud.

**Comment by dejj** [comment]
This seems to take forever.
I moved my owncloud folder to sdcard using:

    cd /mnt/sdcard
    cp -R owncloud /Removable/MicroSD/owncloud
    mv owncloud owncloud.bkup
    su
    mkdir owncloud
    mount -o bind /Removable/MicroSD/owncloud owncloud

**Comment by RonGee** [comment]
Any chance of getting a copy of Ixtalo's version? No sd card support is a total deal-breaker for me.

**Comment by guettli** [comment]
I ask myself if a native android client app is needed at all.

I use FolderSync since some weeks since it can sync to an external sd card. Works nice - I am happy.

Which features does the native app have, which you don't get by using the web interface with your mobile device?

**Comment by terokorp** [comment]
+1

**Comment by guettli** [comment]
@terokorp  what was the +1 for?

please implement this feature

OR

My though: Why is this app needed at all? A web interface and WebDAV sync is all you need.

**Comment by mthmulders** [comment]
@guettli, I personally use the 'share to' feature quite a bit. It allows me to directly upload a picture / video from the Gallery in my phone to my ownCloud server. I don't see how you would do that using the mobile web interface.

**Comment by terokorp** [comment]
@guettli Sorry, I didint read whole discussion, just few at beginning. my 1+ was "please implement this feature".
My device has just 11G for user and owncloud used most of it (used most of it before moving files manually to sd card). Every time when I add new music/audiobooks I have to manually move files to SD card after download.

I have tried few WebDAW sync clients, but none of them had working selective sync. I also use owncloud instant upload and share to ownCloud. This ownCloud client is almost perfect for me, only thing which I need more is reported on this issue.

**Comment by marissacute** [comment]
I would like this feature as well. I want to use owncloud to download my music library to my phone, that's not going to fit on my internal memory.

There must be more use cases like mine where the to be synced data is too large to reasonably go on the phone's internal memory, you can't expect all users to setup WebDAV sync.

**Comment by guettli** [comment]
I create a very related ticket: https://github.com/owncloud/android/issues/1173

Quoting:
{{{


What does the development roadmap of owncloud-android look like?

Where is the focus of the developers today, tomorrow, ...?

What are the next steps?

What will be done later?
}}}

**Comment by tobiasKaminsky** [comment]
There are just too few man power for this.
(And I personally do not have an external sd card, so therefore this issue has low priority for me [note: I am not an official developer and doing this in my spare time).

So either try to support by developing code or there is a bounty hunter: https://www.bountysource.com/teams/owncloud/issues?tracker_ids=11489

**Comment by guettli** [comment]
@tobiasKaminsky  thank you for the hint. I donated 20$ to get **this external sd card issue**.
Unfortunately I saw no way to tell this inside the donation.

**Comment by tobiasKaminsky** [comment]
I do not know the work flow exactly, but this should be the site to start with:
https://www.bountysource.com/issues/541829-configure-download-folder-external_sd

**Comment by tobiasKaminsky** [comment]
(As time changes and therefore new devices are purchased, I am working on this).
A really simple and awful solution is already working: you have to type the path manually and the app does not bother if the folder already exists and just writes into it.

So, for having this fully working
- check if folder exists: warn
- after changing path: move all files  --> is this needed? All files have to be copied and then deleted.

Anything else missing?

**Comment by daemoniac** [comment]
Hello together,

I have a little different problem: My owncloud syncs the pictures, but not the movies. I have a brand new Sony Xperia Z5 - the pictures are safed in internal memory, the videos (because of the size of 4k movie material) on the external sd card.

As written: pictures sync, movies not. 

So it would be great if it would be possible to configure both path manual in the android app.
Thanks so far!

**Comment by tobiasKaminsky** [comment]
I doubt that this has something to do with the location of the owncloud folder itselfs.
You mean instantupload? If so, please open a new issue.

**Comment by przybylski** [comment]
@tobiasKaminsky are you still working on it?
I have some spare time now and would like to grab that ;)

**Comment by tobiasKaminsky** [comment]
@przybylski  I will commit my changes tonight into a branch in this repository, so that you can work on it.

**Comment by tobiasKaminsky** [comment]
@przybylski Here it is: #1258
If you have any question please let me know!

Todo is:
- allow folder choosing
- check if folder exists: warn
- after changing path: move all files --> is this needed? All files have to be copied and then deleted.

**Comment by przybylski** [comment]
@tobiasKaminsky Thanks!
this feature progress can be tracked at my fork: https://github.com/przybylski/android/tree/externalSD

**Comment by przybylski** [comment]
This feature is now ready to merge in https://github.com/owncloud/android/pull/1276
however it requires this to work: https://github.com/owncloud/android-library/pull/91

**Comment by marneu** [comment]
Seconding the request:
Unfortunately the App uses the internal sdcard storage, thus making it on a 16Gb hard to survive where I do have a 128Gb external SD card.
To overcome the barrier it would be nice to do it like Titanium Backup and serve an option to store the synced data on the extSDcard device. Just an idea...

**Comment by tobiasKaminsky** [comment]
@marneu As you can see @przybylski has implemented this feature and it is right now in the beta version.
So there is no need in putting a "+1" reply.
Regarding you proposal about the option: it is already there and was discussed...Please try to search before posting.

**Comment by marneu** [comment]
Thanks Tobias, is there a way to participate in the beta?

**Comment by przybylski** [comment]
@marneu you can download APKs from here https://github.com/owncloud/android/tree/beta/apks/

**Comment by marneu** [comment]
Great: Do i need the uninstall the app before pushing the apk?

**Comment by tobiasKaminsky** [comment]
No. It is possible to install it alongside the official app.
The beta version is also available at f-droid.
But please only use just one instant uploader as otherwise it may come to strange behaviour.

**Comment by mhoess** [comment]
Hi,
syncing to a sd card folder does not work on android 4.x (think because of authorizatization issues for writing on SD Card)
Can somebody help?

**Comment by marneu** [comment]
Did you try SDFix (nextapp.sdfix)?

**Comment by tobiasKaminsky** [comment]
The app should work without any other app required.
I do not know if Andrid 4.x behaves different than >5 regarding the storage on sdcards...

**Comment by EMCP** [comment]
I tried to follow the link towards https://github.com/owncloud/android/blob/beta/apks/owncloud-beta-20151202.apk , inside ES File Manager and in my own browser.  Getting a 'cannot parse' error message on install attempts.

**Comment by tobiasKaminsky** [comment]
Normally this is working the way you described it.
You can also install it via fdroid.

**Comment by bes1002t** [comment]
It would be necessary to implement this feature. I have just 16GB internal storage so I need it soon.

**Comment by przybylski** [comment]
@bes1002t this feature is in testing in beta, you can check is it working for you

**Comment by bes1002t** [comment]
ah ok thank you :)

**Comment by Technical-Angel** [comment]
I'm using ownCloud beta 20151202.

I cannot change the storage path at all. I created a owncloud folder on the SD card and I get Error: While migrating.

When trying to change the folder to internal storage, I get an error log screen. 

I have about 30GB that I'm trying to drop down to the tablet, and I really want to help make this feature work!!! :)

Samsung Galaxy Tab 4, Android version 5.0.2

**Comment by marneu** [comment]
Did you load with f-droid?
You should use the lastest beta and try to switch back to internal and then to external again, this should do the trick.

**Comment by Technical-Angel** [comment]
I uninstalled the official app before I started doing anything with the beta. 

I just removed the beta I had installed and installed 20160120 using f-droid. I previously installed the beta using the built in APK installer. I now only have the option of default or the SD card, but when switching to SD, I still get the error: while migrating message. 

Log file is empty. 

And, as an aside, am I barking up the wrong tree here? I want ownCloud to do on Android what it does on PC, automatically compare the files on the server to the files on the device and synchronize them. I would like for this to happen without any user interaction. 

While going over documentation and settings, it doesn't appear that that is the case. Am I missing something or am I looking at the wrong program?

**Comment by tobiasKaminsky** [comment]
>And, as an aside, am I barking up the wrong tree here? I want ownCloud to do on Android what it does on PC, automatically compare the files on the server to the files on the device and synchronize them. I would like for this to happen without any user interaction.

>While going over documentation and settings, it doesn't appear that that is the case. Am I missing something or am I looking at the wrong program?

This is (currently) not working, but it will be implemented sooner or later. The previous step (manually syncing folders) will be soon in official app. 
(Please correct me if I am wrong @davivel)

**Comment by meduz** [comment]
This is a very nice feature but I also get the "error: while migrating message" message. I am on android 4. Is there any way to see what is going wrong?

**Comment by przybylski** [comment]
@meduz Only via logfile. Because of android system fragmentation it takes some time to provide reliable implementation for all distributions. It would be nice to know that kind of phone you own.

**Comment by meduz** [comment]
Thanks for the answer  @przybylski. 
Below is the information for my device . Unfortunately I could not find a way to root it.

************ DEVICE INFORMATION ***********
Brand: samsung
Device: v1awifi
Model: SM-P900
Id: KOT49H
Product: v1awifixx

************ FIRMWARE ************
SDK: 19
Release: 4.4.2
Incremental: P900XXUANI1

**Comment by przybylski** [comment]
@meduz log file should be stored on your device, in the same place where files are.

**Comment by ksyz** [comment]
This looks like related issue. [SD Card on Android 5 (Lollipop)](https://metactrl.com/docs/sdcard-on-lollipop/)

**Comment by codethief** [comment]
Any updates on this?

**Comment by EMCP** [comment]
@codethief the feature is available in beta

**Comment by cbernet** [comment]
Hi all, this feature would really be great to have. My use case is to set up a home cloud to manage the family files (lots of music, videos, photos). Therefore, I do need a cloud service that supports sd cards on android phones. The lack of support of external sd cards is the only thing driving me away from owncloud at the moment. That's a pity because apart from that, owncloud does look like the best solution.

**Comment by gasparakos** [comment]
I installed owncloud-beta-20160606.apk and working as expected, downloaded files stored on selected external storage. This is on android 4.0.4 (Sony Xperia), removable SD card mounted on /mnt/ext_card. Thank You for your work!

**Comment by ralf1070** [comment]
> Am 31.07.2016 um 08:57 schrieb gasparakos <notifications@github.com>:
> 
> I installed owncloud-beta-20160606.apk and working as expected, downloaded files stored on selected external storage. This is on android 4.0.4 (Sony Xperia), removable SD card mounted on /mnt/ext_card. Thank You for your work!

Any chance we see this in mainline anytime soon?

**Comment by davivel** [comment]
Sorry, but with our current bandwidth, I wouldn't expect this in mainline before winter.

**Comment by zoff99** [comment]
i wonder how anybody is using this without ability to choose folder?
why are other features even developed while this is not fixed?

**Comment by davivel** [comment]
@zoff99 , other people has other priorities different than this. For example, there are devices without SD card support, and their users don't care about a feature like this.

We try to make happy as many people as possible, but we can't satisfy everybody. Of course, we could have done it wrong, maybe this should have been addressed before. It's difficult to tell objectively.

What I can grant you is that we work really hard every day on improve the app, and that our first priority is providing value to the users.

**Comment by zoff99** [comment]
i am happy that the app exists at all

**Comment by teadur** [comment]
+1 this really needed feature and should be get to main asap, whats the holdup if its already in beta ?

**Comment by marneu** [comment]
@taedur: Just a question:
You are using your cam on the smartphone, pictures go to DCIM, right?
What would be the benefit, if you could configure it to, lets say owncloud?

So why should the destination folder be named is a specific way?

**Comment by EMCP** [comment]
@marneu I'm not understanding your question but I'll give you my usecase

I have an NVidia Shield Tablet running the sound system of my car.. hooked up to an external monitor + AUX .  I wanted owncloud app to stick the files on the SD Card so I could get ~100GB of space to place music videos on myown cloud.  When I pull up to a wifi hotspot, I can sync the contents of my SD-Card with my owncloud.

As it stood, owncloud had hardcoded to use my tablets internal memory (16GB) .. and severely limited the usefulness of owncloud for this project.

**Comment by zoff99** [comment]
@EMCP use this version as a workaround, where you can select you card.
!use at your own risk!

https://circleci.com/gh/zoff99/owncloud_android/17#artifacts/containers/0

https://17-67236307-gh.circle-artifacts.com/0/tmp/circle-artifacts.rVJ6sn0/owncloud_android_circleci_e8071b367bdf1db7bb567dfe7cc3599ca74dfedc.apk

**Comment by zoff99** [comment]
why they won't merge (for years) is totally beyond me :-(
it's just 1 click

**Comment by marneu** [comment]
@ECMP I'm using the former beta, loaded from f-droid, and this one does the trick - using a sd-card.
What I missunderstood was, I thought the name of the resulting folder is the prob.

**Comment by meduz** [comment]
I tested the nexcloud beta client (while still using my owncloud clients) and for the first time, the storing of files on the external card effectively worked without a glitch. (my 2-years old sdcard was eagerly waiting for this!)

**Comment by teadur** [comment]
@marneu My use case is basicaly the same as @EMCP 
I have loads of data in owncloud (music), that i want to sync to my sony z5c external sdcard so i can listen to my media on the move, not only behind pc.

**Comment by zoff99** [comment]
+1 nextcloud beta from fdroid works fine

**Comment by davivel** [comment]
Hi, @przybylski . 

May I ask why did you self-assign and then closed the issue? This is not done yet. 

Thanks

**Comment by DeepDiver1975** [comment]
@przybylski please also remove your claim on bountysource - the pr is not merged yet - https://www.bountysource.com/issues/541829-configure-download-folder-external_sd THX

**Comment by przybylski** [comment]
@DeepDiver1975 I would prefer not to remove the claim. My reasoning is:
This feature were ready over a year ago and many users are reporting that it has been working for them for a long time, regardless is it owncloud or nextcloud, which is using the same implementation.
The argument of ignoring this pull request is that there isn't enough "workforce", but I have a feeling that is strictly personal reasons. If it were because of lack of "workforce" then you probably shouldn't ignore my consulting offer which I sent do owncloud couple of times, with no response.
And really the amount of this bounty is a joke and peanuts in what the real worth of the feature is, so I claimed it since I wrote the implementation for it.

**Comment by davivel** [comment]
My reasoning is:

We have a process to review and accept PRs. In this repository, this process includes technical approval by an OC internal developer, quality approval by an OC internal QA engineer, and priority approval by OC management. Might be different, but right now it's that way, for everybody. And this PR didn't go through these stages.

If you want to claim a bounty due to its inclusion in a fork, you should apply to the bounties in the context of that fork.

> The argument of ignoring this pull request is that there isn't enough "workforce", but I have a feeling that is strictly personal reasons. 

I'm sorry you feel that way, but there are no personal reasons behind that. Other PRs by you have been reviewed and approved in this project in the past. The list can be checked by anyone here:

https://github.com/owncloud/android/pulls?utf8=%E2%9C%93&q=is%3Apr%20author%3Aprzybylski%20is%3Amerged

No personal criteria has been applied in any case.

> If it were because of lack of "workforce" then you probably shouldn't ignore my consulting offer which I sent do owncloud couple of times, with no response.

I have no idea of this, but I'll dare to say this is an oversimplification. There are hundreds of legit reasons to leave a consulting offer without response that have nothing to do with personal reasons. 


If you are still interested in get this PR merged, please, rebase it and I'll try to push so that it's considered for higher priority.

**Comment by DeepDiver1975** [comment]
> And really the amount of this bounty is a joke and peanuts in what the real worth of the feature is, so I claimed it since I wrote the implementation for it.

Let me put it this way: as soon as your pr is merged you can easily reclaim the bounty and I'll 200$ on top. Deal?

**Comment by przybylski** [comment]
@DeepDiver1975 deal

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 deal

so - please remove the claim and I'll add the 200$ - thx

**Comment by DeepDiver1975** [comment]
@przybylski fyi: i just removed you from the owncloud orga - you lost all repos access. I'm sorry - but this is not the way this is going to work out. We had a deal and you did break it.

**Comment by zoff99** [comment]
this is ridiculous years of discussion, but no actual progress.

**Comment by przybylski** [comment]
@DeepDiver1975 i tried to remove the claim and i didn't work. I thought that it won't pay out if issue was re-opened. I put back the bounty when I realised it was cashed out.

**Comment by DeepDiver1975** [comment]
so - welcome back then :wink:

**Comment by DeepDiver1975** [comment]
additional 200$ have been placed - happy coding @przybylski

**Comment by przybylski** [comment]
I'll pick it up on Thursday.
@DeepDiver1975  BTW I got your invitation and then I got removed from org, I am guessing its intentional?

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 BTW I got your invitation and then I got removed from org, I am guessing its intentional?

I guess the events have been dispatched in the wrong order - you should be in the org - if not let me know I'll happily add you again.

**Comment by przybylski** [comment]
@DeepDiver1975 so it seems that its a problem with events dispatching.

**Comment by DeepDiver1975** [comment]
@przybylski I just invited you again

**Comment by przybylski** [comment]
progress update: rebase is ongoing, I must make sure that I haven't broke anything else

**Comment by davivel** [comment]
Sorry, but we need to drop issues from milestone 2.4.0 to get it out next to OC 10 server.

This time we'll move topics to ```backlog``` instead of directly to next release, so that we can schedule 2.5.0 properly.

**Comment by guettli** [comment]
The android App FolderSync works nice: See related question here: https://softwarerecs.stackexchange.com/questions/42963/alternative-to-owncloud-which-can-store-data-on-external-sdcard-on-android

**Comment by zoff99** [comment]
7 years, no progress :-(

**Comment by teadur** [comment]
Indeed really sad, seems its time to move to Foldersync and drop using owncloud client for android, becauselt clearly this feature isnt priority of any kind.

**Comment by CorneelDragon** [comment]
Any change that this feature will be put in the 2.6.0?

**Comment by AnrDaemon** [comment]
I have shy of 500 MB free on the device root partition. To update any of the installed applications, I have to wipe the ownCloud sync directory every time.
I feel like the $1 I've paid for the app was too much for the trouble it is causing.

**Comment by sbocconi** [comment]
Still no news in 2018?

**Comment by theScrabi** [comment]
@davigonz do we use the storage framework to save files? With that it would be possible to get access to an external sdcard

**Comment by davigonz** [comment]
> do we use the storage framework to save files? With that it would be possible to get access to an external sdcard

@theScrabi we have a DocumentProvider [here](https://github.com/owncloud/android/blob/master/src/com/owncloud/android/providers/DocumentsStorageProvider.java) but it seems that is not being used at the moment

**Comment by zoff99** [comment]
```
mokasin opened this Issue on Nov 4, 2012
```

please tell me this is not still open?

**Comment by theScrabi** [comment]
@zoff99 Well maybe other things where more important to take care of :D
However we are currently discussing about this, and of course, help is always welcome ;)

**Comment by dubreuia** [comment]
The app is unusable without this, I just refunded my purchase in favor of https://softwarerecs.stackexchange.com/questions/42963/alternative-to-owncloud-which-can-store-data-on-external-sdcard-on-android / https://play.google.com/store/apps/details?id=dk.tacit.android.foldersync.lite

**Comment by nikolayhg** [comment]
The app is also unusable for me without SD Card storage. Uninstalled.

**Comment by switzel** [comment]
In case someone is looking for this: the nextcloud app allows changing the local folder to sd card. The ownCloud one doesn't seem to be actively developed any more.

**Comment by AnrDaemon** [comment]
You're confusing client functionality with server functionality, it seems

**Comment by boltronics** [comment]
@AnrDaemon Assuming that was directed at me, are you implying that the nextcloud Android client supports syncing against ownCloud as well?

**Comment by AnrDaemon** [comment]
I don't see why it would not.

**Comment by davigonz** [comment]
> In case someone is looking for this: the nextcloud app allows changing the local folder to sd card. The ownCloud one doesn't seem to be actively developed any more.

Hi @switzel , @boltronics , @AnrDaemon , it's true we don't currently support sd card but it doesn't mean the app is not actively developed any more, have you checked our [changelog](https://github.com/owncloud/android/blob/master/CHANGELOG.md)? 

We have included a **bunch of new features during the last year** such as video streaming,  gif support, fingerprint and pattern lock, multiple public links per file, stable auto-uploads a.k.a Camera uploads, OAuth2 support, upload picture directly from camera, new wizard, user quota and many, many more. 

I perfectly understand there's some scenarios we are not currently covering but the stability of the ownCloud Android app is one of our strengths and we take that very seriously, betting on quality instead of quantity of features.

Besides, we are putting a big effort on new interesting features, stay tuned!

**Comment by AnrDaemon** [comment]
Perhaps you don't understand, but a user judge the application by the features they are using.
I can't use video streaming or gifs because I can't store anything on the device - I simply don't have free space…

**Comment by theScrabi** [comment]
> Perhaps you don't understand

I think we do. We wont to have a well running and stable application that satisfies our users as well as possible. However implementing something takes time, and as I just said, here fore us and what we see other things, are more important right now. Things such as cleaning the codebase in order not to bad patch features.

 @AnrDaemon if you are desperate on this feature you are highly welcome to support us.

**Comment by theScrabi** [comment]
Please remember on bounty source this issue is open as well:
https://www.bountysource.com/issues/541829-configure-download-folder-external_sd

So if you want to support us you can donate to this issue, maybe someone wants to give it a shot.

**Comment by guettli** [comment]
One work-around: AFAIK  you can use the nextcloud android app and connect it to an owncloud server.

**Comment by PVince81** [comment]
Some use case for this: when you have a big photo collection and want to be able to view it offline.
On some tablet, the internal space is not enough so one must resort to using a micro SD card which enables much more space.

**Comment by GM-Script-Writer-62850** [comment]
Is it possible to do this within the internal storage?
for example i have my music folder on my server i want this accessible offline in a non-default location namely /www/public/music

**Comment by davigonz** [comment]
> Is it possible to do this within the internal storage?
for example i have my music folder on my server i want this accessible offline in a non-default location namely /www/public/music so i can use the media play i made in firefox (running php server on my phone with access via localhost)

Not yet, is not one of our priorities right now but contributions are always appreciated ;)

**Comment by GM-Script-Writer-62850** [comment]
What files should i be looking at if i wanted to try to patch this issue?
What prevents just moving the entire app to the SD card?
How would i go about installing a modded version?

**Comment by jesmrec** [comment]
Version 2.19 of the Android app includes Scoped Storage, and makes no posible to move the content out from the external storage.

**Comment by AnrDaemon** [comment]
Can you please explain that to the general public? Was that an improvement, rejection, or statement of the obvious?

**Comment by jesmrec** [comment]
@AnrDaemon More info here -> https://central.owncloud.org/t/owncloud-2-19-for-android/34952

**Comment by AnrDaemon** [comment]
TL;DR: ownCloud app is now even more useless than before. If before, you could have used it to sync your work between devices, although it took space on your internal flash and you had to repeatedly purge local storage, now all you can do is stare at it. No apps can access the files in the cloud without additional pressure from outside the app (i.e. using some kind of exporter to get the files out of the storage).

**Comment by AnrDaemon** [comment]
You failed to read the terms under which Google requires applications to use scoped storage.
It is required for applications that only store what they create only for their own use. If the result should be used outside the app in question, scoped storage is not the right choice.
Sorry, but… uninstalled. It was good five years waiting for a better app, which apparently will never come.

**Comment by jpfox** [comment]
Agree with @AnrDaemon !
Scoped storage is a mess for synced files. For example, I use ownCloud to sync my password database (a keepass file). When I want to sign in a website with keepass, I have to open ownCloud app, find my keepass file, open it from here and share it with keepass app. And that, each time I want to sign in a website. And it is impossible for keepass to add new password (I can but the file does not sync back to the owncloud server). Keepass provide a useful keyboard to enter easily password, but this widget does not find the keepass database...
Since scoped storage, it is really impossible to use files on the phone.
Before, I was recommanding to use owncloud to my friends and even customers, now, I say that I have no viable solution and let them share all their stuff with Google.

Letting users choose the kind of storage (scoped or external) would be great

**Comment by Helios07** [comment]
We also had dozens of users complaining about use cases not working anymore. They were used to being able to access their files from other apps (pictures, pdf, password safes, etc). I also would like to see this configurable or the old way.

**Comment by riversdev0** [comment]
I just upgraded phones yesterday and along with it came a new Android version (9 to 13) and a new ownCloud App version.  I completely understand that this Scoped Storage change is all in the name of privacy, and that Android is mandating a certain level of adoption, but this has destroyed my use-case for using the App and ownCloud.  I specifically used this as a file transfer mechanism between my phone and PC, and now it no longer performs the only single function for which I used it.  I know enough to say that I don't know how to solve this.  But being unable to access files from other apps makes this app worthless to me.

**Comment by jesmrec** [comment]
ownCloud app implements SAF (document provider), so files inside ownCloud app are available through any app that also implements document provider. You can check this app:  https://play.google.com/store/apps/details?id=com.marc.files. It's only a file browser but it's valid as example that files and folders are accesible out of our app, using the correct and safer method. 

Through that app you can copy and move files from oC to "Download" or any other location of your device, and viceversa.

**Comment by JuancaG05** [comment]
Hey @riversdev0! Thanks for your comment here!
Let's see if I understood correctly: I see you use the app as a transfer tool between your smartphone and your PC, and you say now you cannot perform this task due to scoped storage. If the direction was phone -> PC, I guess you can continue adding files from your app and then downloading them from the web client in your computer, regardless scoped storage. If the direction was PC -> phone, we have some changes in the procedure, for this and the other case you mention. Scoped storage makes your app' files non-accessible from other apps BUT from the document provider implemented by other apps. This is why you can use apps that implement directly this document provider to be able to manage the files from your device (included the ownCloud ones), for example, [Files](https://play.google.com/store/apps/details?id=com.marc.files).

In fact, we are now developing a new feature that adds a direct link to this app from the settings 😁 (https://github.com/owncloud/android/issues/4028):
![new_settings_option](https://user-images.githubusercontent.com/57049315/236390573-9440be44-113c-4d71-96ec-b4177c41d9f9.png)

I hope I could solve your doubts, don't hesitate to ask again if you have any other problem! 🍻

**Comment by riversdev0** [comment]
Hi @jesmrec and @JuancaG05 -  I appreciate your feedback.  Thanks for your the suggestion of using the "Files" app.  My phone had the Samsung "My Files" app already installed, but Samsung apparently doesn't implement the document provider because that app can't see oC as an available file system.  Using the app you suggested, I can now copy the files to a folder on the phone and use them in other apps.  Thank you so much!

**Comment by JuancaG05** [comment]
Hi @riversdev0! Exactly, the Samsung app "My files" (and any other file browser app by any device brand) doesn't implement the Android native file browser, the document provider. That's why we need to look for another solution such as the app we suggested to you. I'm glad you could solve your problem, if you need anything else we're at your disposal 🚀

**Comment by DeepDiver1975** [comment]
🤖 Automated issue triage by Claude Code review agent.

This issue was opened before 2020 and has had no activity since then, so we are
closing it as outdated. If this is still relevant on a current ownCloud release,
please **reopen** (or open a fresh issue) with up-to-date details — thank you\!
